kernel/arch/dreamcast/include/dc/vec3f.h

```diff
@@ -37,408 +41,227 @@ typedef struct vec3f {
-                              "fadd fr3, fr5\n" \
-                              : "+f" (__cx), "+f" (__cz) \
-                              : "f" (__px), "f" (__pz), "f" (__r), "f" (__s) ); \
-        px = __cx; pz = __cz; \
-    }
+static inline vec3f_t vec_rotd_xz(vec3f_t vec, vec3f_t origin, float angle) {
+    return vec_rotr_xz(vec, origin, angle * R_DEG / R_RAD);
+}
 
 /** \brief  Macro to rotate a vector about its origin on the y, z plane.
 
     This macro is an inline assembly operation using the SH4's fast
-    (approximate) math instructions. The return vector is stored into the first
-    vertex parameter: x1, y1, and z1.
-
-    \param  px               The X coordinate of vector to rotate.
-    \param  py               The Y coordinate of vector to rotate.
-    \param  pz               The Z coordinate of vector to rotate.
-    \param  cx               The X coordinate of origin vector.
-    \param  cy               The Y coordinate of origin vector.
-    \param  cz               The Z coordinate of origin vector.
-    \param  r                The angle (in degrees) of rotation.
+    (approximate) math instructions.
+
+    \param  vec             The source vector.
+    \param  origin          The origin vector.
+    \param  angle           The angle (in degrees) of rotation.
 */
-#define vec3f_rotd_yz(px, py, pz, cx, cy, cz, r) { \
-        register float __py __asm__("fr0") = py; \
-        register float __pz __asm__("fr1") = pz; \
-        register float __cy __asm__("fr4") = cy; \
-        register float __cz __asm__("fr5") = cz; \
-        register float __r  __asm__("fr6") = r; \
-        register float __s __asm__("fr7") = R_DEG; \
-        __asm__ __volatile__( \
-                              "fmul fr7, fr6\n" \
-                              "ftrc fr6, fpul\n" \
-                              "fsca fpul, dr6\n" \
-                              "fsub fr4, fr0\n" \
-                              "fsub fr5, fr1\n" \
-                              "fmov fr0, fr2\n" \
-                              "fmov fr1, fr3\n" \
-                              "fmul fr7, fr0\n" \
-                              "fmul fr6, fr1\n" \
-                              "fmul fr6, fr2\n" \
-                              "fmul fr7, fr3\n" \
-                              "fadd fr0, fr4\n" \
-                              "fsub fr1, fr4\n" \
-                              "fadd fr2, fr5\n" \
-                              "fadd fr3, fr5\n" \
-                              : "+f" (__cy), "+f" (__cz) \
-                              : "f" (__py), "f" (__pz), "f" (__r), "f" (__s) ); \
-        py = __cy; pz = __cz; \
-    }
+static inline vec3f_t vec_rotd_yz(vec3f_t vec, vec3f_t origin, float angle) {
+    return vec_rotr_yz(vec, origin, angle * R_DEG / R_RAD);
+}
+
+/** \cond */
+/* Compatibility macros */
+#define vec3f_dot(x1, y1, z1, x2, y2, z2, w) \
+    w = vec_dot((vec3f_t){ x1, y1, z1 }, (vec3f_t){ x2, y2, z2 })
```

Comment:
```suggestion
    (w = vec_dot((vec3f_t){ x1, y1, z1 }, (vec3f_t){ x2, y2, z2 }))
```
If we add an extra set of parenthesis around it, we can use it as a statement expression in conditionals and shit without side-effects or compiler warning about lack of parenthesis.

Reply:
Except that you woudn't use that macro in any way that hasn't been done before.
Because it's a *deprecated* macro, and you should be using the `vec_dot` function instead.

Reply:
I guess. It's true, these macros were written like shit previously, but I figured we might as well fix them to not suck while we were at it... But... I guess it's not a big deal since they're deprecated.  Ah. I see you also aren't even including them in the Doxygen too... So that's good.